dart/lib/src/attachment.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:typed_data';
+export 'attachment_extensions/attachment_extensions.dart';
+
+// https://develop.sentry.dev/sdk/envelopes/#attachment
+
+/// Arbitrary content which gets attached to an event.
+class Attachment {
+  Attachment({
+    required this.content,
+    required this.fileName,
+    AttachmentType? type,
+    String? mimeType,
+  })  : mimeType = mimeType ?? 'application/octet-stream',
+        type = type ?? AttachmentType.attachment;
+
+  /// Creates an [Attachment] from a [Uint8List]
+  factory Attachment.fromUint8List(
```

Comment:
`m`: Can we rename `fromUint8List`, `fromIntList` and `fromByteData` to `fromBytes` and use method overloading? Maybe it's also fine to have only one method with `Uint8List`.

Reply:
Dart has no method overloading, so that's not possible :(

There is no single data type which is used to represent bytes, so that's why I added multiple different constructors.
The Dart HTTP library we use, for example uses `List<int>`and `UInt8List`.

Reply:
yeah I'd go with factories for the different types too